src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -22,38 +51,315 @@ const ActorInit En_Daiku_InitVars = {
+            this->unk_25C = this->unk_25C + 1;
+            if (this->unk_25C >= this->unk_258->count) {
+                this->unk_25C = 0;
+            }
+            func_809437C8(this);
+            this->unk_282 = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_26C);
+        }
+    }
+
+    if (this->unk_278 != ENDAIKU_PARAMS_0) {
+        s16 angle = ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y));
+        this->unk_280 = this->actor.yawTowardsPlayer;
+        if ((this->unk_278 == ENDAIKU_PARAMS_1) || (this->unk_278 == ENDAIKU_PARAMS_2) || (angle < 0x2891)) {
+            func_800B8614(&this->actor, globalCtx, 100.0f);
+        }
+    }
+}
+
+void func_80943BC0(EnDaiku* this) {
+    this->unk_28A = 1;
+    this->actionFunc = func_80943BDC;
+}
+
+void func_80943BDC(EnDaiku* this, GlobalContext* globalCtx) {
+    f32 currentFrame = this->skelAnime.animCurrentFrame;
+
+    if ((this->unk_278 == ENDAIKU_PARAMS_2) && (this->unk_284 <= currentFrame)) {
+        if (Rand_ZeroOne() < 0.5f) {
+            func_8094373C(this, 7);
+        } else {
+            func_8094373C(this, 8);
+        }
+    }
+
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        func_801477B4(globalCtx);
+        func_80943820(this);
+    }
+}
+
+void EnDaiku_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
+    s32 pad;
+
+    if (this->unk_27E == 0) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+
+    if ((this->unk_278 == ENDAIKU_PARAMS_0) && (gSaveContext.day == 3) && (gSaveContext.isNight)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->unk_27C != 0) {
+        this->unk_27C--;
+    }
+
+    if (this->unk_27E != 0) {
+        this->unk_27E--;
+    }
+
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Actor_SetHeight(&this->actor, 65.0f);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Math_SmoothStepToS(&this->unk_260, this->unk_266, 1, 3000, 0);
+    Math_SmoothStepToS(&this->unk_25E, this->unk_264, 1, 3000, 0);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+    this->actor.uncullZoneForward = 650.0f;
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80943E18(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDaiku* this = THIS;
+
+    if (limbIndex == 15) {
+        rot->x += this->unk_260;
+        rot->z += this->unk_25E;
+    }
+
+    return 0;
+}
+
+void func_80943E60(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Gfx* D_809440D4[] = {
+        D_060070C0, D_06006FB0, D_06006E80, D_06006D70, NULL,
+    };
```

Comment:
```suggestion
    static Gfx* D_809440D4[] = {
        D_060070C0, D_06006FB0, D_06006E80, D_06006D70,
    };
```
Is this `NULL` padding or is it required?